lib/cache.js

```diff
@@ -33,15 +39,31 @@ exports.clear = () => {
 };
 
 exports.update = () => {
+  let tempFolder = path.join(TEMP_FOLDER, utils.uniqueId());
+
   // Downloading fresh copy
-  return fs.ensureDir(CACHE_FOLDER)
+  return Promise.all([
+    // Create new folder inside temporary stroage
+    fs.ensureDir(tempFolder),
+    fs.ensureDir(CACHE_FOLDER)
+  ])
     .then(() => {
-      return remote.download();
+      // Download and extract cache data to temporary folder
+      return remote.download(tempFolder);
     })
-    .then((tempFolder) => {
+    .then(() => {
+      // Copy data to cache folder
       return fs.copy(tempFolder, CACHE_FOLDER);
     })
     .then(() => {
-      return index.rebuildPagesIndex();
+      return Promise.all([
+        // Remove temporary storage
+        fs.remove(TEMP_FOLDER),
```

Comment:
It seems that you are creating `tempFolder` inside `TEMP_FOLDER`. But you are deleting `TEMP_FOLDER`. This will fail if 2 users are simultaneously updating their caches. Since the tool will delete the parent dir, there might be a race condition where one user's cache dir gets deleted.

I would suggest to delete `tempFolder` instead and keep the parent dir as is.

Reply:
I'm idiot, didn't take concurrency into equation 🤦♂
Good catch, will do as you suggested 👌

Reply:
That's what reviews are for :wink: